Discontinuous timestamps in AUX.continuous data files

Working subject 41, I received this error:
(tye_lab_to_nwb_env) lkeyes@node32:~/Projects/GIT/tye-lab-to-nwb$ python src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_convert_all_sessions.py
  0%|                                                                                                                                                | 0/1 [00:00<?, ?it/s]                                  Source data is valid!
concurrent.futures.process._RemoteTraceback:
"""
Traceback (most recent call last):
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/concurrent/futures/process.py", line 256, in _process_worker
    r = call_item.fn(*call_item.args, **call_item.kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_convert_session.py", line 145, in session_to_nwb
    converter = NeurotensinValenceNWBConverter(source_data=source_data)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/neuroconv/src/neuroconv/nwbconverter.py", line 64, in __init__
    self.data_interface_objects = {
                                  ^
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/neuroconv/src/neuroconv/nwbconverter.py", line 65, in <dictcomp>
    name: data_interface(**source_data[name])
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/neuroconv/src/neuroconv/datainterfaces/ecephys/openephys/openephysdatainterface.py", line 40, in __new__
    return OpenEphysLegacyRecordingInterface(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/neuroconv/src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py", line 42, in __                                  init__
    available_streams = self.get_stream_names(folder_path=folder_path)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/neuroconv/src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py", line 57, in ge                                  t_stream_names
    stream_names, _ = OpenEphysLegacyRecordingExtractor.get_streams(folder_path=folder_path)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/site-packages/spikeinterface/extractors/neoextractors/neobaseextractor.py", line 142, in get_streams
    neo_reader = get_reader(cls.NeoRawIOClass, **neo_kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/site-packages/spikeinterface/extractors/neoextractors/neobaseextractor.py", line 14, in get_reader
    neo_reader.parse_header()
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/site-packages/neo/rawio/baserawio.py", line 179, in parse_header
    self._parse_header()
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/site-packages/neo/rawio/openephysrawio.py", line 118, in _parse_header
    assert np.all(diff == RECORD_SIZE), \
AssertionError: Not continuous timestamps for 104_AUX1.continuous. Maybe because recording was paused/stopped.
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_convert_all_sessions.py", line 92, in <module>
    parallel_convert_sessions(
  File "/nadata/snlkt/home/lkeyes/Projects/GIT/tye-lab-to-nwb/src/tye_lab_to_nwb/neurotensin_valence/neurotensin_valence_convert_all_sessions.py", line 83, in parallel_con                                  vert_sessions
    future.result()
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/home/lkeyes/anaconda3/envs/tye_lab_to_nwb_env/lib/python3.11/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
AssertionError: Not continuous timestamps for 104_AUX1.continuous. Maybe because recording was paused/stopped.

--- a/src/tye_lab_to_nwb/ast_ecephys/convert_session.py
+++ b/src/tye_lab_to_nwb/ast_ecephys/convert_session.py
@@ -3,6 +3,9 @@ from pathlib import Path
 from typing import Optional, Dict
 from warnings import warn
 
+from importlib.metadata import version
+from packaging.version import Version
+
 from nwbinspector import inspect_nwb
 from nwbinspector.inspector_tools import format_messages, save_report
 
@@ -59,7 +62,11 @@ def session_to_nwb(
     conversion_options = dict()
 
     # Add Recording
-    source_data.update(dict(Recording=dict(folder_path=str(ecephys_recording_folder_path), stream_name="Signals CH")))
+    recording_source_data = dict(folder_path=str(ecephys_recording_folder_path), stream_name="Signals CH")
+    if Version(version("neo")) > Version("0.12.0"):
+        recording_source_data.update(ignore_timestamps_errors=True)
+
+    source_data.update(dict(Recording=recording_source_data))
     conversion_options.update(dict(Recording=dict(stub_test=stub_test)))
 
     # Add Sorting (optional)
@@ -183,7 +190,7 @@ if __name__ == "__main__":
 
     # For faster conversion, stub_test=True would only write a subset of ecephys and plexon data.
     # When running a full conversion, use stub_test=False.
-    stub_test = False
+    stub_test = True
 
     # Run conversion for a single session
     session_to_nwb(

--- a/src/tye_lab_to_nwb/ast_ecephys/convert_session.py
+++ b/src/tye_lab_to_nwb/ast_ecephys/convert_session.py
@@ -190,7 +190,7 @@ if __name__ == "__main__":
 
     # For faster conversion, stub_test=True would only write a subset of ecephys and plexon data.
     # When running a full conversion, use stub_test=False.
-    stub_test = True
+    stub_test = False
 
     # Run conversion for a single session
     session_to_nwb(

--- a/src/tye_lab_to_nwb/ast_ecephys/requirements.txt
+++ b/src/tye_lab_to_nwb/ast_ecephys/requirements.txt
@@ -1,3 +1,3 @@
--e git+https://github.com/catalystneuro/neuroconv@main#egg=neuroconv
+-e git+https://github.com/catalystneuro/neuroconv@add_ignore_timestamps_errors_to_interface#egg=neuroconv
 neuroconv[openephys,plexon,sleap,video]
 openpyxl>=3.1.2

